File: ParteC#/ProyectoProgramacion3/ProyectoProgramacion3Model/Model/reportes/Reporte.cs
namespace ProyectoProgramacion3Model.Model.reportes;
public abstract class Reporte
{
    public int idReporte { get; set; }
    public DateOnly fechaGeneracion { get; set; }
    public DateOnly fechaInicioFiltro { get; set; }
    public DateOnly fechaFinFiltro { get; set; }
    public Reporte(int idReporte, DateOnly fechaGeneracion, DateOnly fechaInicioFiltro, DateOnly fechaFinFiltro)
    {
        this.idReporte = idReporte;
        this.fechaGeneracion = fechaGeneracion;
        this.fechaInicioFiltro = fechaInicioFiltro;
        this.fechaFinFiltro = fechaFinFiltro;
    }
    public Reporte() 
    { 
    }
}

File: backend/ParteJAVA/ProyectoPro3Modelo/src/main/java/pe/edu/pucp/proyectopro3/modelo/reportes/Reporte.java
package pe.edu.pucp.proyectopro3.modelo.reportes;

import java.util.Date;

public abstract class Reporte {
    private int idReporte;
    private Date fechaGeneracion;
    private Date fechaInicioFiltro;
    private Date fechaFinFiltro;

    public Reporte(int idReporte, Date fechaGeneracion, Date fechaInicioFiltro, Date fechaFinFiltro) {
        this.idReporte = idReporte;
        this.fechaGeneracion = fechaGeneracion;
        this.fechaInicioFiltro = fechaInicioFiltro;
        this.fechaFinFiltro = fechaFinFiltro;
    }

    public Reporte() {

    }

    public int getIdReporte() {
        return idReporte;
    }

    public void setIdReporte(int idReporte) {
        this.idReporte = idReporte;
    }

    public Date getFechaGeneracion() {
        return fechaGeneracion;
    }

    public void setFechaGeneracion(Date fechaGeneracion) {
        this.fechaGeneracion = fechaGeneracion;
    }

    public Date getFechaInicioFiltro() {
        return fechaInicioFiltro;
    }

    public void setFechaInicioFiltro(Date fechaInicioFiltro) {
        this.fechaInicioFiltro = fechaInicioFiltro;
    }

    public Date getFechaFinFiltro() {
        return fechaFinFiltro;
    }

    public void setFechaFinFiltro(Date fechaFinFiltro) {
        this.fechaFinFiltro = fechaFinFiltro;
    }
}
